distribution/docs/src/main/jdocs/content/_security-framework/realms-contents.adoc

```diff
@@ -0,0 +1,109 @@
+
+|org.codice.ddf.security.handler.guest.GuestHandler
+|Handler that allows guest user access via a guest user account.
+The guest account credentials are configured via the org.codice.ddf.security.claims.guest.GuestClaimsHandler.
+The `GuestHandler` also checks for the existence of basic auth credentials or PKI credentials that might be able to override the use of the guest user.
+
+|org.codice.ddf.security.filter.login.LoginFilter
+|This filter runs immediately after the WebSSOFilter and exchanges any authentication information found in the request with a Subject via Shiro.
+
+|org.codice.ddf.security.filter.authorization.AuthorizationFilter
+|This filter runs immediately after the `LoginFilter` and checks any permissions assigned to the web context against the attributes of the user via Shiro.
+
+|org.apache.shiro.realm.AuthenticatingRealm
+|This is an abstract authenticating realm that exchanges an `org.apache.shiro.authc.AuthenticationToken` for a `ddf.security.Subject` in the form of an `org.apache.shiro.authc.AuthenticationInfo`
+
+|ddf.security.realm.sts.StsRealm
+|This realm is an implementation of `org.apache.shiro.realm.AuthenticatingRealm` and connects to an STS (configurable) to exchange the authentication token for a Subject.
+
+|ddf.security.service.AbstractAuthorizingRealm
+|This is an abstract authorizing realm that takes care of caching and parsing the Subject's `AuthorizingInfo` and should be extended to allow the implementing realm to focus on making the decision.
+
+|ddf.security.pdp.realm.AuthZRealm
+|This realm performs the authorization decision and may or may not delegate out to the external XACML processing engine. It uses the incoming permissions to create a decision.
+However, it is possible to extend this realm using the ddf.security.policy.extension.PolicyExtension interface.
+This interface allows an integrator to add additional policy information to the PDP that can't be covered via its generic matching policies.
+This approach is often easier to configure for those that are not familiar with XACML.
+Note that no `PolicyExtension` implementations are provided out of the box.
+
+|org.codice.ddf.security.validator.*
+|A number of STS validators are provided for X.509 (BinarySecurityToken), UsernameToken, SAML Assertion, and ${branding} custom tokens.
+The ${branding} custom tokens are all `BinarySecurityTokens` that may have PKI or username/password information as well as an authentication realm (correlates to JAAS realms installed in the container).
+The authentication realm allows an administrator to restrict which services they wish to use to authenticate users.
+For example: installing the `security-sts-ldaplogin` feature will enable a JAAS realm with the name "ldap".
+This realm can then be specified on any context using the Web Context Policy Manager.
+That realm selection is then passed via the token sent to the STS to determine which validator to use.
```

Comment:
We should probably drop a blurb in here stating that using the SAML Web SSO Identity Provider for authentication will ignore any realm settings and simply use all of the JAAS realms that are configured.

Reply:
added.